### ISSUE TITLE
Duplication Glitch: Double Kill

### ISSUE THREAD

**Issue by Shibva** [SammySemicolon/Malum-Mod#175]
Im not sure about this fully but here's what I have:

When killing things with the scythe, I've noticed that at times I get 2 death messages. (this occurs when I kill my pet by accident and 2 broken death messages appear) 
![image](https://user-images.githubusercontent.com/38770780/224302423-3d176d77-63d9-428d-82c9-03d519b96dbd.png)
(was able to dig this up)

The second is that with nametag tweaks that it drops twice as many nametags (for all I know this could be a bug on their end but it seems to be niche to this issue in particular)

Third: When this double-kill occurs; the mod Dommestic Innovation's respawn mechanic for tamed pets (if in the instance you did what I did) will respawn the same pet twice. Again this could be a issue on their side but its also niche to this situation; and also coraltes to the entity dying twice.

Do you have any idea whats going on here?

**Comment by Shibva** [comment]
Its only been observed wtih the Soulstained Scythe; though if the effects persist with  the Iron Scythe or the Tyrving are unknown as of this time

**Comment by SammySemicolon** [comment]
I've known about it for some time. I'm not sure how to go about fixing it, might just tackle it on in a later version of mc

**Comment by Shibva** [comment]
I wondered if it has something to do with the different types of mechanics that you could use do it kill someone with a scythe with; you know the fact that I can sweep and do magic damage or something to do with a focus strike with that one pedant on


It seems only apply to win the players, swinging it as I have not seen anything of the sort happening when I use it as a boomerang with the rebound enchantment 

Depending on how it's coded, maybe add a conditional statement to see if the entity is already dead. By the looks of it is applying to different damage types at the same time which tells me that's probably it has something to do with that. If this is the case, then I wonder if putting some sort of conditional statement to see if the end of these alive or dead before applying the second damage type would fix the issue. Then again, I'm not exactly sure how all this works, but that's just a thought.

**Comment by Shibva** [comment]
Ok more awake now


I wonder if it is being caused because of a code fault.

I feel that sweeping damage should NOT be applied to the entity you hit directly because sweep damage is lower than direct damage

A fix would probs be implementing something to prevent sweeping damage from effecting the main entity you hit 

Could also expain why there may be extra damage to the base weapon then intended (because sweep is dealing more on top of the main damage to the entity)

**Comment by Shibva** [comment]
Furthermore is probably best to have all damage types be applied as one value of its instead being multiple different damages

As in just get the sum of the damage types and then apply them all as one damage source


Depends on what you did; just giving ideas to what might be at fault and ways to go about fixing it

**Comment by SammySemicolon** [comment]
> Furthermore is probably best to have all damage types be applied as one value of its instead being multiple different damages

Ain't possible, magic damage and physical damage respect armor values differently

**Comment by Shibva** [comment]
Right; because magic if I am thinking correctly here is like RAW damage [bypass armor; like how Jeb_ coded vindicators to ignore armor]

My best guess is that it may have todo wtih how the overall damage scheme is coded or that there's something else running that shouldn't be; like how I believe the damage from sweep is being applied to the main entity you attacked when it most likely should not be (that would also give it more damage that what might be intended)


Maybe if you have the time and dont mind you can explain to me how the scythe and its damage system works and everything that goes with it; maybe then I (or others that may view this) might be able to notice something that was probably overlooked.

Though I am certain on one thing: it has something todo wtih how it was made to deal damage and/or hurt entities specifically wtih the Soulstained Scythe [if not just scythes in general]

I will note areas that may be worth looking into/elaborating on

- The way the Soulstained Scythe handes dealing damage [it would be notable if there's a difference between the normal and soulstained one]
- The alteration of their attack behavior when the user is wearing either the Necklace of the Narrow Edge or Necklace of the Hidden Blade
- Whatever Voodoo is

**Comment by Shibva** [comment]
Ok, I managed to reproduce wtih wtih a soulsteel knife as well
